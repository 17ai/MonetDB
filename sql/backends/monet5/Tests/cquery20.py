import sys

try:
    from MonetDBtesting import process
except ImportError:
    import process

def server_stop(s):
    out, err = s.communicate()
    sys.stdout.write(out)
    sys.stderr.write(err)

def client(input):
    c = process.client('sql', stdin = process.PIPE, stdout = process.PIPE, stderr = process.PIPE)
    out, err = c.communicate(input)
    sys.stdout.write(out)
    sys.stderr.write(err)

script1 = '''\
create temporary stream table sta (a int);
'''

script2 = '''\
create stream table stb (a int);
'''

script3 = '''\
select count(*) from streams;
'''

script4 = '''\
drop table stb;
'''

def main():
    s = process.server(args = [], stdin = process.PIPE, stdout = process.PIPE, stderr = process.PIPE)
    client(script1)
    client(script3)
    client(script2)
    client(script3)
    server_stop(s)
    s = process.server(args = [], stdin = process.PIPE, stdout = process.PIPE, stderr = process.PIPE)
    client(script3)
    client(script4)
    client(script3)
    server_stop(s)

if __name__ == '__main__':
    main()
