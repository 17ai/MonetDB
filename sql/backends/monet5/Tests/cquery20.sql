-- create a continuous query and find it

create stream table cqtbl(i integer);

-- the hello example
create continuous procedure cqfoo(v integer)
begin
	insert into cqtbl values(v);
end;
select * from functions where name = 'cqfoo';

-- continuous procedures can always be called directly
-- this simplifies debugging as well
-- Main limitation 
call cqfoo(123);
select * from cqtbl;

start continuous cqfoo(321);
select * from cqtbl;

stop continuous cqfoo(321);
--stop continuous cqfoo;
select * from cqtbl;

drop procedure cqfoo;
drop table cqtbl;

