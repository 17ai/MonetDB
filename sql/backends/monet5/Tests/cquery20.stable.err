stderr of test 'cquery20` in directory 'sql/backends/monet5` itself:


# 23:19:50 >  
# 23:19:50 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=34472" "--set" "mapi_usock=/var/tmp/mtest-6809/.s.monetdb.34472" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/trails//Linux/var/MonetDB/mTests_sql_backends_monet5"
# 23:19:50 >  

# builtin opt 	gdk_dbpath = /export/scratch1/mk/trails//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 34472
# cmdline opt 	mapi_usock = /var/tmp/mtest-6809/.s.monetdb.34472
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /export/scratch1/mk/trails//Linux/var/MonetDB/mTests_sql_backends_monet5
# cmdline opt 	gdk_debug = 536870922

# 23:19:50 >  
# 23:19:50 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-6809" "--port=34472"
# 23:19:50 >  


# 23:19:50 >  
# 23:19:50 >  "Done."
# 23:19:50 >  

