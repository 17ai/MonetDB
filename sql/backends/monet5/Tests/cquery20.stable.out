stdout of test 'cquery20` in directory 'sql/backends/monet5` itself:


# 23:19:50 >  
# 23:19:50 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=34472" "--set" "mapi_usock=/var/tmp/mtest-6809/.s.monetdb.34472" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/trails//Linux/var/MonetDB/mTests_sql_backends_monet5"
# 23:19:50 >  

# MonetDB 5 server v11.28.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.588 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2017 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:34472/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-6809/.s.monetdb.34472
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/Timetrails module loaded

Ready.

# 23:19:50 >  
# 23:19:50 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-6809" "--port=34472"
# 23:19:50 >  

#create stream table cqtbl(i integer);
#create continuous procedure cqfoo(v integer)
#begin
#	insert into cqtbl values(v);
#end;
#select * from cqtbl;
% sys.cqtbl # table_name
% i # name
% int # type
% 3 # length
[ 123	]
#select * from cqtbl;
% sys.cqtbl # table_name
% i # name
% int # type
% 3 # length
[ 123	]
[ 321	]
#drop procedure cqfoo;
#drop table cqtbl;

# 23:19:50 >  
# 23:19:50 >  "Done."
# 23:19:50 >  

